backend/src/main/java/c/team/quiz/QuizController.java

```diff
@@ -0,0 +1,52 @@
+package c.team.quiz;
+
+import c.team.account.UserAccountService;
+import c.team.quiz.exception.QuizNotFoundException;
+import c.team.quiz.model.CreateQuizRequest;
+import c.team.quiz.model.CreateQuizResponse;
+import c.team.quiz.model.GetQuizzesResponse;
+import c.team.quiz.model.Quiz;
+import c.team.security.model.UserPrincipal;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("quiz")
+@AllArgsConstructor
+public class QuizController {
+
+    private final UserAccountService userAccountService;
+    private final QuizService quizService;
+
+    @PostMapping
+    public ResponseEntity<CreateQuizResponse> createQuiz(@AuthenticationPrincipal UserPrincipal user, @RequestBody @Valid CreateQuizRequest request) {
```

Comment:
W jaki sposób współdziała teraz to @AuthenticationPrincipal UserPrincipal z requestami HTTP? Trzeba coś tam więcej dorzucać (po stronie frontendu)? Jak to działa?

Reply:
Nie, przy sprawdzaniu tokena dodaje sie do kontekstu `@AuthenticationPrincipa`, i można potem z niego wyciągnąć dane przy requestach